### ISSUE TITLE
Database corrupted , buckets and its objects are not shown by s3 ls

### ISSUE THREAD

**Issue by nehasharma5** [noobaa/noobaa-core#6675]
Environment info
[root@api.ns.cp.fyre.ibm.com ~]# oc version
Client Version: 4.7.13
Server Version: 4.7.13
Kubernetes Version: v1.20.0+df9c838
[root@api.ns.cp.fyre.ibm.com ~]# noobaa version
INFO[0000] CLI version: 5.9.0
INFO[0000] noobaa-image: noobaa/noobaa-core:master-20210719
INFO[0000] operator-image: noobaa/noobaa-operator:5.9.0
[root@api.ns.cp.fyre.ibm.com ~]#

Actual behavior
Data base corrupted , buckets and its objects are not shown by s3 ls (Bucket name - neha and object : testfile)

Expected behavior
No Database corruption should be done , bucket and its object should be listed by s3 ls

Steps to reproduce
Install noobaa and start a copy object operation into a bucket
Once copy is finished , start another copy operation
While copy operation is going on shutdown the node on which noobaa is installed (I have only endpoint pod scheduled on that node , no other noobaa pode)
Start the node
Delete DB  pod
App Node:
[root@fyreauto-x-app1 ~]# s3 ls
urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host 's3-noobaa.apps.ns.cp.fyre.ibm.com'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
[root@fyreauto-x-app1 ~]# s3 cp testfile s3://neha
urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host 's3-noobaa.apps.ns.cp.fyre.ibm.com'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
upload failed: ./testfile to s3://neha/testfile An error occurred (NoSuchKey) when calling the CreateMultipartUpload operation: The specified key does not exist.
[root@fyreauto-x-app1 ~]# s3 ls
urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host 's3-noobaa.apps.ns.cp.fyre.ibm.com'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
[root@fyreauto-x-app1 ~]# s3 mb s3://new-buck
urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host 's3-noobaa.apps.ns.cp.fyre.ibm.com'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
make_bucket failed: s3://new-buck An error occurred (AccessDenied) when calling the CreateBucket operation: Access Denied

Inf node:
[root@api.ns.cp.fyre.ibm.com ~]# oc get pod -o wide
NAME                                               READY   STATUS        RESTARTS   AGE   IP             NODE                         NOMINATED NODE   READINESS GATES
noobaa-core-0                                      1/1     Running       0          73m   10.254.3.167   master1.ns.cp.fyre.ibm.com   <none>           <none>
noobaa-db-pg-0                                     1/1     Running       0          60m   10.254.4.17    master0.ns.cp.fyre.ibm.com   <none>           <none>
noobaa-default-backing-store-noobaa-pod-62daf8d7   0/1     Terminating   0          38m   <none>         master2.ns.cp.fyre.ibm.com   <none>           <none>
noobaa-endpoint-565dbbd667-gfzt2                   1/1     Running       0          74m   10.254.4.14    master0.ns.cp.fyre.ibm.com   <none>           <none>
noobaa-operator-6d54447bc5-hr7sb                   1/1     Running       1          19h   10.254.3.136   master1.ns.cp.fyre.ibm.com   <none>           <none>
[root@api.ns.cp.fyre.ibm.com ~]#

Storage cluster
[root@storage-x-master neha]# ls -lrt
total 3
drwxr-x---. 3 2001 2002 4096 Jul 20 04:15 buck-1
drwxr-x---. 3 2001 2002 4096 Jul 22 02:36 my-buck
drwxr-x---. 3 2001 2002 4096 Jul 22 04:26 neha
[root@storage-x-master neha]#

[endp.log](https://github.com/noobaa/noobaa-core/files/6867479/endp.log)
[must-gather.local.2961412144701386214.tar.gz](https://github.com/noobaa/noobaa-core/files/6867480/must-gather.local.2961412144701386214.tar.gz)

**Comment by nimrod-becker** [comment]
@nehasharma5 Is this different than #6678 ?

**Comment by nehasharma5** [comment]
During #6678 I found two issues one of which is #6675 but the steps to attempt both are different #6675 and #6678 is different

**Comment by jackyalbo** [comment]
@nehasharma5 How did you decide the DB got corrupted? Not sure I follow. from the logs it seems the DB is aware of a bucket called neha at least. But it does seems like the directory was deleted from the storage:
```
2021-07-23 06:49:52.349234 [PID-8/TID-8] [L1] FS::FSWorker::Begin: CheckAccess _path=/nsfs/fs1/neha
2021-07-23 06:49:52.349260 [PID-8/TID-18] [L1] FS::FSWorker::Execute: CheckAccess _path=/nsfs/fs1/neha _uid=2001 _gid=2002 _backend=GPFS
2021-07-23 06:49:52.349290 [PID-8/TID-18] [L1] FS::FSWorker::Execute: CheckAccess _path=/nsfs/fs1/neha  took: 0.011431 ms
2021-07-23 06:49:52.349395 [PID-8/TID-8] [L1] FS::FSWorker::OnError: CheckAccess _path=/nsfs/fs1/neha  error.Message()=No such file or directory
Jul-23 6:49:52.349 [Endpoint/8]    [L0] core.sdk.bucketspace_nb:: _has_access_to_nsfs_dir: failed [Error: No such file or directory] { code: 'ENOENT' }
```

Can you check the data is still accessible also from inside the endpoint pod and attach to this issue.
Also please share the endpoint.yaml - So we can see if the correct pvc is attached to it.

**Comment by nehasharma5** [comment]
@jackyalbo s3 ls does not show any any bucket and endpoint too . There is no issue from SC , SC have bucket and objects [root@storage-x-master neha]# cd neha
[root@storage-x-master neha]# ls
testfile  testfile1
[root@storage-x-master neha]#
Data is not accessible from endpoint pod also

**Comment by nehasharma5** [comment]
@jackyalbo This issue has been discussed with IBM Storage team , there is an issue with CSI as Static pv data not accessible by noobaa endpoint. I have raised another issue in CSI so closing this issue
